### ISSUE TITLE
Make numpad . enter a decimal separator in input fields

### ISSUE THREAD

**Issue by Gymnasiast** [OpenRCT2/OpenRCT2#15589]
### Describe the feature

The numpad . key is meant to input a decimal separator appropriate for the locale at hand. In OpenRCT2, it always inputs a dot, which makes it impossible to use it to type prices if the locale the user has uses commas. Rather it should use whether the current language has set on the "decimal separator" string ID.

### Feature type

Quality of life

**Comment by bhoward27** [comment]
I'd like to work on this issue.

**Comment by tupaschoal** [comment]
Alright

**Comment by schuylermartin45** [comment]
Curious if there's any update on this, I noticed while playing today (x86, Linux).

Probably needs a separate issue, but I also noticed that the numeric entry doesn't work on some of the inputs in the option menus. I understand this is probably intentional to limit fractional scaling values, but it's a little annoying when changing the screen scaling moves the +/- buttons. Maybe it could round to the nearest accepted value, like the price dialog box?

Thanks for the hard work on this, everyone. This project is awesome!

**Comment by Gymnasiast** [comment]
> Probably needs a separate issue, but I also noticed that the numeric entry doesn't work on some of the inputs in the option menus. I understand this is probably intentional to limit fractional scaling values, but it's a little annoying when changing the screen scaling moves the +/- buttons. Maybe it could round to the nearest accepted value, like the price dialog box?

This is not intentional, but the spinner/stepper input widget does not come with this functionality out-of-the-box. It has to be added manually for every widget, unless we overhaul our window system.

**Comment by matheusvb3** [comment]
I'd like to work on this!

**Comment by tupaschoal** [comment]
Go on!

**Comment by matheusvb3** [comment]
So after doing some testing I'm a little uncertain about this. When the user presses the numbpad dot it gets replaced with the appropriate decimal separator for the game's current language, useful for those who have only the dot key in their keyboard numpads, but for my native language's layout (Portuguese Brazil ABNT), which have both dot and comma keys, this results in both keys printing a dot when the game's language is English (since the SDLK_KP_PERIOD key corresponds to numpad comma in my layout, go figure), which is obviously not desired. I've tried using SDLK_KP_DECIMAL but that doesn't work either. I think the only way to solve this in a satisfactory way would be to check the user's current keyboard layout which would require some more work. Furthermore, the only two solutions I've found to set out accomplishing what OP requested are inelegant, I could either use a global variable in `TextComposition.cpp` which is hacky or add a new a new attribute to the `TextComposition` class which seems overkill. What are your thoughts everyone?

**Comment by tupaschoal** [comment]
Yeah, I see what you're saying, some ABNT2 Brazilian keyboards can have both comma and dot, in any case I wouldn't mind both printing the appropriate decimal separator if only we could define that we are in a context where it is needed (like when inputting ticket prices), but not on the chat or console, for example

**Comment by matheusvb3** [comment]
I tried looking into it, but unfortunately I don't think there's any way to differentiate between numeric and text input boxes right now.... Of course, numeric fields have spinner widgets, but there's no way to know that in `TextComposition.cpp` as far as I can tell, which is where you actually get to type in the numbers you want. Given how frequently they're used, I think creating a `NumericComposition.cpp` file for number inputs could be a solution, and we could make it behave similarly to HTML numeric input fields, but then I'm not sure if it could break plugins or have other side effects..... I'd really need some more input (haha) from the developers into this...

**Comment by tupaschoal** [comment]
Any say @Gymnasiast ?
